File: task3/task3a_b/src/main/java/com/task3a_b/Book.java
package com.task3a_b;

public class Book extends Product{
    private String bookIBAN;
    private String bookGenre;
    private String bookAuthor;

    public Book(){

    }

    public Book(int id, String name, double price, String category, int volume, String IBAN, String genre, String author){
        super(id, name, price, category, volume);
        bookIBAN = IBAN;
        bookGenre = genre;
        bookAuthor = author;
    }

    @Override
    public void display_product_info(){
        super.display_product_info();
        System.out.println("Book IBAN: "+getIBAN());
        System.out.println("Book Genre: "+getGenre());
        System.out.println("Book Author: "+getAuthor());
    }

    // Setters
    public void setIBAN(String IBAN){
        bookIBAN = IBAN;
    }

    public void setGenre(String genre){
        bookGenre = genre;
    }

    public void setAuthor(String author){
        bookAuthor = author;
    }

    // Getters
    public String getIBAN(){
        return bookIBAN;
    }

    public String getGenre(){
        return bookGenre;
    }
    
    public String getAuthor(){
        return bookAuthor;
    }
}

File: task1/task1b/Book.cpp
#include "Book.h"
#include <iostream>

Book::Book(){}

Book::Book(int id, const string& name, double price, const string& category, int volume, const string& i, const string& g, const string& a): Product(id, name, price, category, volume), IBAN(i), genre(g), author(a) {}

void Book::display_product_info() {
    Product::display_product_info();
    cout << "IBAN: " << IBAN << endl;
    cout << "Book Genre: " << genre << endl;
    cout << "Author: " << author << endl;
}

// Setters
void Book::setIBAN(const string& i){
    IBAN = i;
}

void Book::setGenre(const string& g){
    genre = g;
}

void Book::setAuthor(const string& a){
    author = a;
}

// Getters
string Book::getIBAN() const{
    return IBAN;
}

string Book::getGenre() const{
    return genre;
}

string Book::getAuthor() const{
    return author;
}